warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py

```diff
@@ -227,10 +229,30 @@ def send_commands_from_testdata(self, testdatafile, **args):
                     result, response = new_obj_session._send_command_retrials(
                         details_dict, index=i, result=result,
                         response=response, system_name=td_sys)
-                    rspRes, response_dict = new_obj_session._get_response_dict(
-                        details_dict, i, response, response_dict)
-                    result = (result and rspRes) if "ERROR" not in (
-                                result, rspRes) else "ERROR"
+                    rspRes, resp_key_list = new_obj_session._get_response_dict(
+                        details_dict, i, response, resp_key_list)
+
+                    if resp_key_list:
+                        for resp in resp_key_list[i].keys():
+                            # session id is formed from the system name and session name.
+                            session_id = \
+                             self.get_session_id_for_resp_ref(details_dict, resp_key_list[i],
+                                                              resp, i, system_name,
+                                                              session_name, key)
+                            td_resp_dict = get_object_from_datarepository(str(session_id))
+                            # checks if title_row value is not in td_resp_dict
```

Comment:
I'm finding it difficult to understand the flow here, can you please explain the flow of what is being done after getting the session-id

Reply:
In line 271 it checks for title_row value not in td_resp_dict, 
Consider a td block which has 3 commands: 
td1 --> c1, c2, c3 and its respective sys tag s1, s2, s1

In the above scenario, if the response reference is stored for td1_c1 then it need to check the particular session exists. After that it needs to check if the title/row is present, if not it needs to be added to the td_resp_dict. 
The same is done for td1_c2 since it has a different sys 's2'.
In td1_c3 it will check for the title/row value which would be available since the title/row was updated to the td_resp_dict of sys 's1' for 'c1' command

Reply:
Thank you for the explanation .. good one.

I have some questions the way td_resp dict is handled, please go through the scenario below and can you add a test to make sure it works fine.

td1  : 
cmd1   resp_req = y (default)
cmd2   resp_req = n
cmd3   resp_req = y (default)
cmd4   resp_req = n 

First command:
- on executing the first command, a empty td_resp_dict is sent to 'update_resp_ref_to_repo'
- since resp_req = y, you will grab the td_resp_dict from data repo, update it and return it back. 
I assume the dictionary is passed by reference it should update both data repo and funciton local copy as well But we need to have a test case to test this.

continue.. 
second command:
- for the second command you will send the updated td_resp_dict to 'update_resp_ref_to_repo'

- Since resp_ref=n you will return whatever was received.

third command:
- send updated td_resp_dict to 'update_resp_ref_to_repo' 
- since resp_ref = y you will grab the td_resp_dict from data_repo .. we need to make sure that the update we made for command one in line-277 actuall updated the data repo as well.
Because if it did not then in tis case we will lose command-1 response..

Can you please add a test with this scenario and make sure all command responses are updated in data_repo and in function local copy as well